torch/_dynamo/output_graph.py

```diff
@@ -696,6 +699,20 @@ def compile_and_call_fx_graph(self, tx, rv, root):
         cg.make_call_generated_code(name)
         return cg.get_instructions()
 
+    @property
+    def placeholders(self) -> Iterator[fx.Node]:
+        r = []
+        for node in self.graph.nodes:
+            if node.op == "placeholder":
+                r.append(node)
+                continue
+            break
+        return r
```

Comment:
list comprehension is your friend
```suggestion
     return [node for node in self.graph.nodes if node.op == "placeholder"]
```

Reply:
No, this is explicitly bad because you will iterate over the entire graph. As placeholders are guaranteed to all at the beginning, an explicit loop will terminate once we finish the placeholders.